src/deprecations.jl

```diff
@@ -0,0 +1,69 @@
+#####
+##### upgrades/deprecations
+#####
+
+# For some functions below, we use `Base.depwarn`.
+#
+# In these cases, we don't use `@deprecate` because the method we're "redirecting to" for the deprecation path
+# (e.g. `_deprecated_read_table`) is different than the method that we're suggesting downstream callers use
+# instead (`Legolas.read`).
+
+using Base: depwarn
+
+function _deprecated_read_table(io_or_path, schema=nothing)
+    table = Legolas.read_arrow(io_or_path)
+    schema isa Legolas.Schema && Legolas.validate(table, schema)
+    return table
+end
+
+function read_signals(io_or_path; validate_schema::Bool=true)
+    depwarn("`Onda.read_signals(io_or_path)` is deprecated, use `Legolas.read(io_or_path)` instead", :read_signals)
+    return _deprecated_read_table(io_or_path, validate_schema ? Legolas.Schema("onda.signal@1") : nothing)
+end
+export read_signals
+
+function read_annotations(io_or_path; validate_schema::Bool=true)
+    depwarn("`Onda.read_annotations(io_or_path)` is deprecated, use `Legolas.read(io_or_path)` instead", :read_annotations)
+    return _deprecated_read_table(io_or_path, validate_schema ? Legolas.Schema("onda.annotation@1") : nothing)
+end
+export read_annotations
+
+@deprecate materialize Legolas.materialize
```

Comment:
`@deprecate` exports by default, so you might need
```suggestion
@deprecate materialize Legolas.materialize false
```
to avoid exporting this when it wasn't before. (Same below.)